osu.Game/Overlays/Mods/ModState.cs

```diff
@@ -32,9 +32,21 @@ public class ModState
         public bool PendingConfiguration { get; set; }
 
         /// <summary>
-        /// Whether the mod is currently filtered out due to not matching imposed criteria.
+        /// Whether the mod is currently valid for selection.
+        /// This can be <see langword="false"/> in scenarios such as the free mod select overlay, where not all mods are selectable
+        /// regardless of search criteria imposed by the user selecting.
         /// </summary>
-        public BindableBool Filtered { get; } = new BindableBool();
+        public BindableBool ValidForSelection { get; } = new BindableBool(true);
+
+        /// <summary>
+        /// Determine if <see cref="Mod"/> is valid and can be shown
+        /// </summary>
+        public bool IsValid => MatchingFilter.Value && ValidForSelection.Value;
```

Comment:
This is incredibly confusing naming. `IsValid` next to `ValidForSelection` is inscrutable, and the xmldoc is not helpful. Please rename to `Visible` or something other that actually means something that is distinct from the other flags.